python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -175,8 +175,38 @@ def router_multicast(self) -> None:
 
     @structured_config_contributor
     def hardware_counters(self) -> None:
-        """hardware_counters set based on hardware_counters.features variable."""
-        self.structured_config.hardware_counters = self.inputs.hardware_counters
+        """
+        Set hardware_counters.
+
+        Contributing data sources:
+          - hardware_counters.features variable.
+          - platform_settings.feature_support.hardware_counters fact.
+          - platform_settings.feature_support.hardware_counter_features fact.
+        """
+        if not self.inputs.hardware_counters:
+            return
+        if not self.shared_utils.platform_settings.feature_support.hardware_counters:
+            # Since we use the same data model in eos_cli_config_gen, it would pick up the input vars unless we explicitly set it to null.
+            self.custom_structured_configs.nested.hardware_counters = EosCliConfigGen.HardwareCounters._from_null()
+            return
+        hardware_counters = self.inputs.hardware_counters._deepcopy()
+
+        # Filter different hardware counter features based on the platform supportability
+        hardware_counters.features = hardware_counters.features._filtered(
+            lambda feature: get_v2(
+                self.shared_utils.platform_settings.feature_support.hardware_counter_features,
+                feature.name.replace(" ", "_").replace("-", "_"),
+                # Assume all uncovered/new features are supported
+                default=True,
```

Comment:
In the description we have mentioned `Feature will be ignored on unsupported platforms.` but here we have set `default` as `true`, looks like contradict statements.
May be we can update the description like `Feature will be ignored on unsupported platforms when set to false`.

Reply:
The line you are commenting on is specifically for hardware_counters where the config input model is just a list of strings, so it could easily contain things we don't cover in the feature toggles. So this default is just saying "yes something is supported because we don't know the option"

Reply:
> The line you are commenting on is specifically for hardware_counters where the config input model is just a list of strings, so it could easily contain things we don't cover in the feature toggles. So this default is just saying "yes something is supported because we don't know the option"

Oh yes got that, comment is not just for this line it is for every key where `default: true` and description says `Feature will be ignored on unsupported platforms.`
It will all make sense if we can just make the description little better, may be like I have suggested above.

Reply:
It is really `Feature will be ignored on platforms where this is 'false'`

Reply:
Looks good!